pkg/registry/client.go

```diff
@@ -0,0 +1,177 @@
+package registry
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	orasregistry "oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// OCIRegistryClient storage client for an OCI registry.
+type OCIRegistryClient struct {
+	host        string
+	project     string
+	certFile    string
+	insecure    bool
+	dryRun      bool
```

Comment:
Maybe an idea for the future, but if there were a separate implementation, `DryRunRegistryClient` or the like, then the code doesn't need to branch on the dry run flag (i.e. it's simpler), and is therefore easier to read, and less prone to someone adding a method implementation and forgetting to check the value of the dry run flag and abort before making changes.

Reply:
That was my original intention, but it ended up having a lot of repeated code. It was a complete copy of the client minus the actual call to CopyGraph